### PR TITLE
refactor: use frformat package

### PR DESCRIPTION
**Pull request by Sarrabah**
# Context

The library fr-format has been developed for sharing validation functions between [validata](https://gitlab.com/validata-table) and csv-detective, and to introduce a standard library to validate typical French formats. 

The aim of this PR is to replace custom validation with the implementation of fr-format.

# Refactorings

- code postal et code commune Insee
- code canton, numero departement, code region
- code fantoir
- canton, departement, commune, region, pays
- Latitude_l93, Longitude_l93
- code RNA

# Behavior changes

- After noticing code Fantoir prefixes starting with two letters, we slightly changed the regex to accept this format (eg. ZB12A).
- CodeRNA does not allow 'w' as first letter, only uppercase 'W'.

# Performance

```
Performance report ─=≡Σ((( つ•̀ω•́)つ

Testing table with 100000000 rows

                     without       with     fr-format

"8730"

code_postal          10.27 s       10.62 s
code_fantoir         10.22 s       10.66 s
code_commune         10.27 s       10.47 s

"ABCDE"

code_postal          12.32 s       11.40 s
code_fantoir         12.24 s       11.37 s
code_commune         11.73 s       11.33 s

"12345"

code_postal          11.63 s       11.38 s
code_fantoir         11.23 s       11.05 s
code_commune         11.31 s       10.97 s
```

The differences do not appear to be statistically significant, given the variability between the two executions observed. 

Edit the 29 May 2024

### CODE REVIEW

File: tests/test_file.py [hunk 1]
```diff
@@ -35,7 +35,6 @@ def test_columns_output_on_file():
     assert output["columns"]["STRUCTURED_INFO"]["format"] == "json"
     assert output["columns"]["GEO_INFO"]["python_type"] == "json"
     assert output["columns"]["GEO_INFO"]["format"] == "json_geojson"
-    assert output["columns"]["NUMEPCI"]["format"] == "siren"
```
Comment (Sarrabah) [inline tests/test_file.py:38]: Removed as this assertion was a duplicate of l.33

File: csv_detective/detect_fields/FR/geo/latitude_l93/__init__.py [hunk 2]
```diff
@@ -1,13 +1,20 @@
...
 
 PROPORTION = 0.9
 
 
 def _is(val):
-    '''Renvoie True si val peut etre une latitude en Lambert 93'''
     try:
-        is_float(val) and float(val) >= 6037008 and float(val) <= 7230728
-    except ValueError:
+        if type(val) is float or type(val) is int:
```
Comment (Pierlou) [inline csv_detective/detect_fields/FR/geo/latitude_l93/__init__.py:11]: Maybe `isinstance(val, (float, int))` ?

File: csv_detective/detect_fields/FR/geo/latitude_l93/__init__.py [hunk 3]
```diff
@@ -1,13 +1,20 @@
...
+            return LatitudeL93.is_valid(val)
+
+        elif type(val) is str and is_float(val):
```
Comment (Pierlou) [inline csv_detective/detect_fields/FR/geo/latitude_l93/__init__.py:14]: Same `isinstance(val, str) and is_float(val)` ?

File: csv_detective/detect_fields/FR/geo/longitude_l93/__init__.py [hunk 4]
```diff
@@ -1,13 +1,20 @@
...
 
 PROPORTION = 0.9
 
 
 def _is(val):
-    '''Renvoie True si val peut etre une longitude en métropole'''
     try:
-        is_float(val) and float(val) >= -357823 and float(val) <= 1313633
-    except ValueError:
+        if type(val) is float or type(val) is int:
```
Comment (Pierlou) [inline csv_detective/detect_fields/FR/geo/longitude_l93/__init__.py:11]: `isinstance`

File: csv_detective/detect_fields/FR/geo/longitude_l93/__init__.py [hunk 5]
```diff
@@ -1,13 +1,20 @@
...
+            return LongitudeL93.is_valid(val)
+
+        elif type(val) is str and is_float(val):
```
Comment (Pierlou) [inline csv_detective/detect_fields/FR/geo/longitude_l93/__init__.py:14]: `isinstance`

**Review by Pierlou — COMMENTED** [review]
HUGE rework, thanks a lot! 🎉

File: csv_detective/detect_fields/FR/geo/latitude_l93/__init__.py (hunk 2 above)
Comment (Sarrabah) [reply csv_detective/detect_fields/FR/geo/latitude_l93/__init__.py:11]: Yes, it's a good idea to replace the method "type()", thank you.

**Review by Pierlou — APPROVED** [review]
Very nice, thanks!

**Comment by Pierlou** [comment]
Seems like the checks are having a hard time, could you add an empty commit to trigger them again please?

**Comment by Sarrabah** [comment]
Yes of course !

**Comment by Pierlou** [comment]
Finally merged with the PR using a branch : https://github.com/datagouv/csv-detective/pull/87 ✅

**Pull request closed without merging**